### PR TITLE
Use PorterDuff blend mode when below Android 29

### PR DESCRIPTION
**Pull request by bparrishMines**
Engine fix for https://github.com/flutter/flutter/issues/104686.

As of now, this is solved at the plugin level by switching to Hybrid Composition when the view has a transparent/translucent background:  https://github.com/flutter/plugins/pull/6063

This should hopefully remove the need to do this. I tested it locally and the previous rendering issue was fixed when this flag was used.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/android/io/flutter/plugin/platform/PlatformViewWrapper.java [hunk 1]
```diff
@@ -308,7 +309,7 @@ public void draw(Canvas canvas) {
         if (Build.VERSION.SDK_INT >= 29) {
           surfaceCanvas.drawColor(Color.TRANSPARENT, BlendMode.CLEAR);
         } else {
-          surfaceCanvas.drawColor(Color.TRANSPARENT);
+          surfaceCanvas.drawColor(Color.TRANSPARENT, PorterDuff.Mode.CLEAR);
```
Comment (dnfield) [inline shell/platform/android/io/flutter/plugin/platform/PlatformViewWrapper.java:312]: These blocks can be simplified by using drawPaint and setting the color and blend mode on the paint. That's all the Android impl is doing here anyway

Comment (dnfield) [reply shell/platform/android/io/flutter/plugin/platform/PlatformViewWrapper.java:312]: You then won't have the branch here or need extra tests for different api levels

Comment (bparrishMines) [reply shell/platform/android/io/flutter/plugin/platform/PlatformViewWrapper.java]: Are you referring to this method? https://developer.android.com/reference/android/graphics/Paint#setBlendMode(android.graphics.BlendMode).

Unfortunately, it also requires API 29.

Although, I agree that we don't need both methods. The `PorterDuf` blend mode isn't deprecated so I think we could just switch to that.

**Review by dnfield — APPROVED** [review]

File: shell/platform/android/io/flutter/plugin/platform/PlatformViewWrapper.java (hunk 1 above)
Comment (dnfield) [reply shell/platform/android/io/flutter/plugin/platform/PlatformViewWrapper.java]: Ahh ok, I didn't realize that. This looks good to me though.

**Pull request merged**